### ISSUE TITLE
Error in dorado trim: "MN tag and sequence length are not in sync"

### ISSUE THREAD

**Issue by simondrue** [nanoporetech/dorado#631]
# The issue:

I get the following error when using `dorado trim` version 5.3.0 (I do not get the error in `dorado` 5.1.0) :

```
[2024-02-16 13:29:10.152] [debug] > adapter/primer trimming threads 33, writer threads 3
[2024-02-16 13:29:10.158] [info] > starting adapter/primer trimming
[2024-02-16 13:29:10.158] [debug] Processed 0 reads
[2024-02-16 13:29:10.158] [debug] Total reads processed: 0
[2024-02-16 13:29:10.158] [trace] Checking adapter/primer LSK109
[2024-02-16 13:29:10.158] [trace] Checking adapter/primer LSK110
[2024-02-16 13:29:10.158] [trace] Detected front interval adapter/primer - LSK109_FWD
[2024-02-16 13:29:10.158] [trace] Checking adapter/primer PCR_PSK_rev1
[2024-02-16 13:29:10.159] [trace] Checking adapter/primer PCR_PSK_rev2
[2024-02-16 13:29:10.159] [trace] Checking adapter/primer cDNA_VNP
[2024-02-16 13:29:10.159] [trace] Checking adapter/primer cDNA_SSP
[2024-02-16 13:29:10.159] [trace] Checking adapter/primer PCS110_forward
[2024-02-16 13:29:10.159] [trace] Checking adapter/primer PCS110_reverse
[2024-02-16 13:29:10.159] [trace] Checking adapter/primer RAD
terminate called after throwing an instance of 'std::runtime_error'
  what():  MN tag and sequence length are not in sync.
```
# Steps to reproduce the issue:

I've managed to find this single read that reproduces the error:

```
@HD	VN:1.6	SO:unknown
@PG	ID:basecaller	PN:dorado	VN:0.5.3+d9af343	CL:dorado basecaller --verbose --modified-bases 5mCG_5hmCG 6mA --no-trim /faststorage/project/MomaReference/BACKUP/nanopore/models/dorado_models/dna_r10.4.1_e8.2_400bps_hac@v4.3.0 test_data_mini/
@PG	ID:samtools	PN:samtools	PP:basecaller	VN:1.17	CL:samtools view -N read_id.txt -h test_mod.bam
@RG	ID:20ec702665c209fcfe07215c4f0432bc7a635fac_dna_r10.4.1_e8.2_400bps_hac@v4.3.0	PU:PAS63454	PM:PC24B154	DT:2024-01-12T10:04:45.764+00:00	PL:ONT	DS:basecall_model=dna_r10.4.1_e8.2_400bps_hac@v4.3.0 modbase_models=dna_r10.4.1_e8.2_400bps_hac@v4.3.0_5mCG_5hmCG@v1,dna_r10.4.1_e8.2_400bps_hac@v4.3.0_6mA@v2 runid=20ec702665c209fcfe07215c4f0432bc7a635fac	LB:Q03A03284D_N513-008	SM:Q03A03284D_N513-008
9c68f0ee-84d9-441d-aa52-f81c9e931b70	4	*	0	0	*	*	0	0	ATTATGTACTGTACTTGGTAGTGTTATGTCCTGTACTTGGTTCAGTTACGTATTGCTAAGGTTAAGCATAGTTCTGCATAATGG	###$'))%$&%&&%*(&(+%%%&(+)+)(%#$%&'&(*)**+87:::162234<30///89<?7:9:8877:::B6611,+*(*	qs:i:15	du:f:0.2668	ns:i:1334	ts:i:10	mx:i:1	ch:i:1622	st:Z:2024-01-13T07:57:56.950+00:00	rn:i:204222	fn:Z:PAS63454_pass_df99ae21_20ec7026_11111.pod5	sm:f:-751.172	sd:f:0.00798761	sv:Z:pa	dx:i:0	RG:Z:20ec702665c209fcfe07215c4f0432bc7a635fac_dna_r10.4.1_e8.2_400bps_hac@v4.3.0	MN:i:84	MM:Z:A+a.,0,0,0,0,0,0,0,1,0;C+h?,6;C+m?,6;	ML:B:C,215,199,27,200,21,250,130,177,21,33,16
```

## Run environment:
- Dorado version: 5.3.0
- Dorado command: `trim`
- Operating system: Linux

**Comment by malton-ont** [comment]
Hi @simondrue,

Thank you very much for reporting this.

It looks like that particular read is being trimmed such that there are no remaining modifications in the `MM/ML` tags and dorado is then failing to update the `MN` tag. I've been able to reproduce the issue, and a fix will be made available in the next release of dorado.

**Comment by tijyojwad** [comment]
Hi @simondrue - this is now fixed in the v0.6.0 release.
